hw/ip/otbn/dv/otbnsim/sim/insn.py

```diff
@@ -148,10 +148,12 @@ def execute(self, model: OTBNModel) -> None:
 
         if self.wb_variant > 0:
             if self.wb_variant == 1:
-                model.set_wr_halfword(self.wrd, acc, self.wrd_hwsel)
-                acc = acc >> 128
-            elif self.wb_variant == 2:
                 model.state.wreg[self.wrd].set(acc)
+            elif self.wb_variant >= 2:
+                # set_wr_halfword expects 0 in upper 128 bits
+                acc_lower = acc & 0xFFFF_FFFF_FFFF_FFFF_FFFF_FFFF_FFFF_FFFF
```

Comment:
Could you write the mask as `(1 << 128) - 1`? Counting F's gets old pretty fast.